foursight_core/react/api/aws_network.py

```diff
@@ -0,0 +1,421 @@
+import boto3
+import copy
+import json
+import re
+from typing import Callable, Optional, Union
+from dcicutils.obfuscation_utils import obfuscate_dict
+from .misc_utils import memoize, sort_dictionary_by_case_insensitive_keys
+
+
+def _filter_boto_description_list(description: dict,
+                                  name: str,
+                                  predicate: Optional[Union[str, re.Pattern, Callable]] = None,
```

Comment:
It's worth noting that `find_association` and `find_associations` in `dcicutils.misc_utils` already does a lot of this search-stuff in a very general way that you've partly reimplemented in a very special-case way. It's possible that using those functions,  you could eliminate the need for some of the detailed special cases here. I won't require that refactoring for this review because we've got other things to do, but you should definitely look at what those can do because given a list of the form [{k1: v1, k2: v2, k3: v3, ...}, ...] you can find 1 or several entries where a given kN matches a given string or predicate. It could easily be extended to handle regular expressions, too. And then we could do stuff like this a different way.

Reply:
I think find_associations (which I do use elsewhere, e.g. envs.py) will not quite fit as in this case the key/values list is a sub-object of the (list of) main objects I'm searching for. Perhaps something more general could be built; I'll add a TODO comment to look into it.